petastorm/pytorch.py

```diff
@@ -308,6 +337,10 @@ def __iter__(self):
         for batch in self._yield_batches(keys):
             yield batch
 
+        # We purposely do not wrap it under a final block, because must finish a full pass of Petastorm DataLoader
+        # before making another pass from the beginning
+        self._in_iter = False
```

Comment:
What would happen if the consumer of batches raises an error? Perhaps would be safer to put `self._in_iter` into a `finally` clause? (would be nice to test this case)